execution/src/posix/execution.cc

```diff
@@ -463,11 +448,26 @@ namespace leatherman { namespace execution {
                                             options[execution_options::inherit_locale]);
         auto envp = to_exec_arg(&variables);
 
+        // Set all open file descriptors above stderr to close on exec
+        auto max_desc_limit = get_max_descriptor_limit();
```

Comment:
Yeah. I don't think we exec in parallel anywhere, but I could add a (global) mutex to guard it. I'm pretty sure we shouldn't do being this stuff in parallel anyway.